src/components/MainPageColumn/MainPageColumn.js

```diff
@@ -12,15 +12,16 @@ const MainPageColumn = ({
     return (
         <div className="mainpage-column">
             <h3>{name}</h3>
-            {tasks.map((task, i) => 
+            {
+                tasks && tasks.map((task, i) => 
                 <Task 
-                    key={i}
                     noRightArrow={noRightArrow}
                     noLeftArrow={noLeftArrow}
                     setAllTasks={setAllTasks}
                     taskStage={task.stage}
                     allTasks={allTasks}
                     task={task}
+                    key={i}
```

Comment:
You can keep index in array as key, but it is not best practice. I think you have unic id property on task item, and it will reduce renders on delete/sort/filter etc.